### PR TITLE
!!!TASK: Add PHP 7.0 scalar type hints to method arguments and return types

### PR DESCRIPTION
**Pull request by albe**
This is breaking because it changes the cache interfaces and all implementations need to add scalar typehints.
The scope should be very limited though, since the amount of custom cache implementations is assumingly low. The most likely break will be with custom `*Privilege` implementations that also override the `getCacheEntryIdentifier()` method.

### CODE REVIEW

**Comment by bwaidelich** [comment]
This seems to break unit tests (https://travis-ci.org/neos/flow-development-collection/jobs/243879407) or is that unrelated?

**Comment by albe** [comment]
That's why it's marked as breaking !!! ;) Nah j/k, that's correct. Will check what happens this weekend.

File: Neos.Cache/Classes/Backend/BackendInterface.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ public function getPrefixedIdentifier($entryIdentifier);
      * @throws \Neos\Cache\Exception\InvalidDataException if $data is not a string
```
Comment (kdambekalns) [inline Neos.Cache/Classes/Backend/BackendInterface.php]: With `$data` being declared as `string` that exception can never be thrown anymore, no? Might be the case for other places as well…

Comment (albe) [inline Neos.Cache/Classes/Backend/BackendInterface.php]: Good catch

**Comment by albe** [comment]
Rebased, removed the obsolete InvalidDataException hints and this should fix the test :crossed_fingers:

**Comment by kitsunet** [comment]
We should actually provide a core migration for updating implementations of `CacheAwareInterface`

**Comment by albe** [comment]
How would that migration look? A regex replacement for the method name `getCacheEntryIdentifier`? Isn't that a bit too generic?

**Comment by kitsunet** [comment]
@albe I think we have code to detect implementations via lexer, so we could prefilter classes implementing the interface and then regex for the method, that should be quite safe. Can try to do that now.

**Comment by kitsunet** [comment]
Ok, this is not what I had in mind but that would require change to the migration system. This should be good enough to catch 99% of all cases. Will merge in 24h if travis is happy and no objections.

**Pull request merged**